### PR TITLE
UAR-826: Update trusts skeleton journey added to web

### PR DESCRIPTION
**Pull request by conuirchgov**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-826

### Change description

Added 4 new pages to the Update journey in relation to Trusts
- Interrupt page
- Tell us about the Trust page
- Individuals or Entities involved page
- OE associated Trusts page

The Update - Review BO/MO page now conditionally links to:
- Trust interrupt page - when flag `FEATURE_FLAG_ENABLE_UPDATE_TRUSTS_30062023` is enabled
- Update - Check your answers page - when flag `FEATURE_FLAG_ENABLE_UPDATE_TRUSTS_30062023` is disabled

The Update - Check your answers page now conditionally links back to:
- OE associated Trusts page - when flag `FEATURE_FLAG_ENABLE_UPDATE_TRUSTS_30062023` is enabled
- Update - Review BO/MO page - when flag `FEATURE_FLAG_ENABLE_UPDATE_TRUSTS_30062023` is disabled

The `isActiveFeature` middleware has been added to each of the new skeleton pages, to ensure they're only accessible when the flag is enabled.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/routes/index.ts [hunk 1]
```diff
@@ -460,6 +464,46 @@ router.route(config.UPDATE_BENEFICIAL_OWNER_TYPE_URL)
...
+  )
+  .get(updateTrustsIndividualsOrEntitiesInvolved.get)
+  .post(updateTrustsIndividualsOrEntitiesInvolved.post);
+
+router.route(config.UPDATE_TRUSTS_ASSOCIATED_WITH_THE_OVERSEAS_ENTITY_URL)
+  .all(
+    isFeatureEnabled(config.FEATURE_FLAG_ENABLE_UPDATE_TRUSTS),
+    authentication,
+    companyAuthentication,
+    navigation.hasUpdatePresenter,
```
Comment (corybudd-ch) [inline src/routes/index.ts:502]: just a note and thinking out loud - we're going to have to remember to update the navigation.hasUpdatePresenter nav checks on each of these when we get to individual tickets and start adding stuff to appData.

**Review by corybudd-ch — APPROVED** [review]

File: views/update/update-trusts-submission-interrupt.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+{% block backLink %}
+  {% include "includes/back-link.html" %}
+{% endblock %}
+
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds-from-desktop">
+      <h1 class="govuk-heading-xl">You now need to submit trust information</h1>
+      <form method="post">{% include "includes/save-and-continue-button.html" %}</form>
```
Comment (stephanie-ch) [inline views/update/update-trusts-submission-interrupt.html:15]: Should be just a continue button?

File: test/controllers/update/update.trusts.submission.interrupt.controller.spec.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      expect(resp.header.location).toEqual(config.UPDATE_TRUSTS_TELL_US_ABOUT_IT_URL);
+    });
+
+    test('when feature flag is off, 404 is returned', async () => {
+      mockIsActiveFeature.mockReturnValue(false);
+
+      const resp = await request(app).post(config.UPDATE_TRUSTS_SUBMISSION_INTERRUPT_URL);
+
+      expect(resp.status).toEqual(404);
+      expect(resp.text).toContain("Page not found");
```
Comment (stephanie-ch) [inline test/controllers/update/update.trusts.submission.interrupt.controller.spec.ts:83]: minor - but there's a PAGE_NOT_FOUND_TEXT variable

**Review by stephanie-ch — APPROVED** [review]

File: views/update/update-trusts-submission-interrupt.html (hunk 2 above)
Comment (conuirchgov) [reply views/update/update-trusts-submission-interrupt.html]: Updated :)

File: test/controllers/update/update.trusts.submission.interrupt.controller.spec.ts (hunk 3 above)
Comment (conuirchgov) [reply test/controllers/update/update.trusts.submission.interrupt.controller.spec.ts]: Now using it

**Pull request merged**